src/Interpolations.jl

```diff
@@ -269,6 +269,15 @@ Base.:(/)(wi::WeightedArbIndex, x::Number) = WeightedArbIndex(wi.indexes, wi.wei
 
 ### Indexing with WeightedIndex
 
+struct InterpGetindex{N,A<:AbstractArray{<:Any,N}}
+    coeffs::A
+    InterpGetindex(itp::AbstractInterpolation) = InterpGetindex(coefficients(itp))
+    InterpGetindex(A::AbstractArray) = new{ndims(A),typeof(A)}(A)
+end
+Base.@propagate_inbounds Base.getindex(A::InterpGetindex{N}, I::Vararg{Union{Int,WeightedIndex},N}) where {N} =
+    interp_getindex(A.coeffs, I, ntuple(d->0, Val(N))...)
+
+# TODO: should we drop the following injection?
```

Comment:
Can you elaborate? Is this fully obsolete now?

Reply:
`Base._getindex` is an internal api anyway. Since `Interpolations.jl` itself no-longer use it. It makes sense to me to dop it.